### DISCUSSION TITLE
从excel中复制公式到univer sheet中，公式没有被复制

### DISCUSSION THREAD

**Discussion by fddqfddq** [dream-num/univer#2285] [Q&A | 常见问题]
从excel中复制带公式的单元格，没有复制到univer sheet中，还得重新在初始化后又去range.setValues

**Comment by fddqfddq** [comment]
大量公式加载就会很慢。页面会卡死，有没有好的办法呢？

**Reply by Dushusir** [reply]
加载大量公式确实是耗时操作，目前只能依靠公式引擎内部的缓存优化。可以提供有个具体的脱敏Excel文件给我们测试一下吗？

**Reply by fddqfddq** [reply]
@Dushusir ，感谢关注 ，Excel文件如下，麻烦抽空帮忙看下

这是其中一个带公式excel，在500行以内初始化比较快,随着行数增加，会比较慢，vue页面会出现等待提示。
如果有2000行左右的数据，univerjs在不带公式时初始化还很快，带公式后初始化就非常慢了。

**Comment by Dushusir** [comment]
> 从excel中复制带公式的单元格，没有复制到univer sheet中，还得重新在初始化后又去range.setValues

公式无法从Excel中复制出来，这是Excel的限制

**Reply by fddqfddq** [reply]
同一个excel文件我复制到Luckysheet里是可以的复制出公式的，在univerjs里就不行。
我临时是通过先复制到luckysheet里生成json data 然后再初始化到univerjs里。

**Reply by maying0505** [reply]
是怎么做到excel里复制到luckysheet，能带着公式复制的

**Reply by AnAiYi** [reply]
luckysheet中的导入功能，能带着公式的, 但是luckysheet 和 univerjs 数据格式不一致
